core/services/keeper/registry_synchronizer_sync.go

```diff
@@ -5,12 +5,17 @@ import (
 	"math/big"
 	"sync"
 
+	"github.com/ethereum/go-ethereum/common"
 	"github.com/pkg/errors"
 
 	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/ethkey"
 	"github.com/smartcontractkit/chainlink/core/utils"
 )
 
+// keeperList is a local lookup map that is refreshed every Registry sync.
+var keeperList = map[int32]map[common.Address]int32{}
+var keeperListmutex = &sync.RWMutex{}
```

Comment:
this mutex is getting shared b/t all instances of reg syncer / upkeep executer. Ex writes from one will block writes from another, even through they touch different data. Probably better to scope this on the job somehow.

I think `keeperList` is technically a memory leak since it is always in scope and doesn't get cleaned up when a job gets deleted.

Reply:
Why not just store the keepers list in DB instead of memory? When updating values in DB we can do a simple join

Reply:
considering the added complexity we're facing of syncing these in memory, I'm leaning more towards storing the info in the db also. I don't think we need two mechanisms for sharing the same information b/t goroutines.

Reply:
ya i thought it was going to make it simpler but that obv isnt so so im going to switch it